File: go/words/numerical/numerical.go
package numerical

func Fibonacci(n int) int {
	if n < 0 {
		return -1
	} else if n < 2 {
		return n
	}
	acc2 := 0
	acc1 := 1
	for n > 2 {
		acc2, acc1 = acc1, acc1+acc2
		n--
	}
	return acc1 + acc2
}

func Factorial(n int) int {
	if n < 0 {
		return -1
	}
	acc := 1
	for n > 1 {
		acc *= n
		n--
	}
	return acc
}

func GCD(a, b int) int {
	for b != 0 {
		a, b = b, a%b
	}
	return a
}

func LCM(a, b int) int {
	return (a / GCD(a, b)) * b
}

File: _experimental/_scala/words/src/main/scala/numerical.scala
class Numerical {}

object Numerical {
	def fibonacci(n: Int): Int = {
		if (n < 0) return -1
		if (n < 2) return n
		var acc1 = 1
		var acc2 = 0
		for (_ <- 3 to n) {
			val temp = acc1
			acc1 = acc1 + acc2
			acc2 = temp
		}
		acc1 + acc2
	}

	def factorial(n: Int): Int = {
		if (n < 0) return -1
		if (n == 0 || n == 1) return 1
		var acc = 1
		for (i <- 2 to n) {
			acc *= i
		}
		acc
	}

	def gcd(a: Int, b: Int): Int = {
		if (b == 0) return a.abs
		gcd(b, a % b)
	}

	def lcm(a: Int, b: Int): Int = {
		(a / gcd(a, b)) * b
	}
}